Feature/added cli option checkboxes

radoslawkoziol opened the pull request:
# Enhanced CLI Modal and Plugin Compatibility Updates

## Changes

### CLI Modal Improvements
![image](https://github.com/user-attachments/assets/ec5f88c9-dcf2-4b96-a52f-24d13a2ca7b3)
- Added dynamic "generation options" checkboxes to the CLI modal
- This enhancement simplifies parameter input for users

### UI Improvements
![image](https://github.com/user-attachments/assets/b396198b-ccba-4b02-8a43-1c4fdf784c5c)
- Added a separator in the menu between "Directory" and "Nest Controller" entries
- Improves menu readability and visual organization

### Plugin Compatibility
- Modified plugin compatibility settings to support future IDE versions
- Changed `pluginUntilBuild` to `253.*` as a working solution
  - Note: According to documentation, `pluginUntilBuild = ` or `pluginUntilBuild = *` should work, but testing showed otherwise
  - The implemented solution (`253.*`) has been verified to work correctly

### Code Quality
- Performed code refactoring to improve maintainability
- Removed `.idea/gradle.xml` file from repository to keep it clean

## Summary
These changes focus on improving user experience with better CLI parameter handling, clearer menu organization, and ensuring the plugin works with future IDE versions.

dinbtechit commented on gradle.properties:
```diff
@@ -4,19 +4,19 @@ pluginGroup = com.github.dinbtechit.jetbrainsnestjs
 # Supported build number ranges and IntelliJ Platform versions -> https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html
 pluginSinceBuild = 223
-pluginUntilBuild = 243.*
+pluginUntilBuild = 253.*
```
Please undo this change. This will cause errors as `253.*` is not a valid version. I just figured out an alternate approach to support future release. I will be releasing a patch for this shortly.

radoslawkoziol replied:
Sure, I can restore the previous version, but as I mentioned before - are you sure it won't work? `verifyPlugin` passes successfully

dinbtechit replied:
I am pretty sure it wont work for the version that is not support yet. I have tried it in the past but I can double check unless Intellij Changed something in their gradle plugin. Gradle is notorious for caching old stuff perhaps the gradle project needs to removed and re-synced Anywho, I have a other option that I am working on to make it support all future releases.

radoslawkoziol replied:
Ok, reverted

dinbtechit commented on gradle.properties:
```diff
@@ -4,19 +4,19 @@ pluginGroup = com.github.dinbtechit.jetbrainsnestjs
 # Plugin Dependencies -> https://plugins.jetbrains.com/docs/intellij/plugin-dependencies.html
 # Example: platformPlugins = com.intellij.java, com.jetbrains.php:203.4449.22
-platformPlugins = PsiViewer:2022.3, com.github.dinbtechit.vscodetheme:1.10.2
+platformPlugins = PsiViewer:2022.3
```
Any specific reason for removing this? This Plugin will not be bundled with the plugin. I added this to do some testing for my VSCode plugin.

radoslawkoziol replied:
Yes, I know it doesn't affect the build in any way, but I didn't have it installed, causing it to throw me errors in the console. I decided that this plugin was not necessary, so, to keep things clean, I removed it from here. I can, of course, restore it ;)

Ideally, I would like to throw some of this stuff (especially `platformType` and `platformVersion`) into some local configuration - `env` or something like that. I personally tested these things on a newer version of WebStorm

dinbtechit commented:
Hi @radoslawkoziol , 

Thank you for submitting a PR. I really appreciate. However, I have some concerns with the UX here. I understand UX is subjected to person to person. The Checkbox style UX adds more confusion than helping with the ease of use. As it now has 2 input points, one via the textfield and the checkbox. Its also making difficult to address the error validation scenarios. Especially the you still have to use the the parameter textfield to input the filename. There is an autocomplete (Ctrl + space or Cmd + Space) that lists all the available options. 

I would definitely up discussing a better UX approach if you want to start a discussion over that and get community feedback before implementing them. Please let me know I can certainly start one.

Additionally, the `pluginUntilBuild` is set to `253.*` which is an invalid build number intellij hasn't released an IDE with that version yet this will cause error in the plugin verification stage of CI/CD. I am working on a more of an automated approach to support all future builds. 

Now that the negatives aside. I do like the idea of adding separator in the context menu. It does provide a neat separation from rest of the menu items. Cheers for that.

Hopefully my above comments make sense. If you agree with the can update the PR accordingly? 🙌

dinbtechit reviewed (changes requested):
Please see above comments.

radoslawkoziol commented:
> Thank you for submitting a PR. I really appreciate. However, I have some concerns with the UX here. I understand UX is subjected to person to person. The Checkbox style UX adds more confusion than helping with the ease of use. As it now has 2 input points, one via the textfield and the checkbox. Its also making difficult to address the error validation scenarios. Especially the you still have to use the the parameter textfield to input the filename. There is an autocomplete (Ctrl + space or Cmd + Space) that lists all the available options.
> 
I was also considering this, but I decided that in this form it works quite efficiently. Maybe I didn’t clarify it in the comment, but the text field and checkboxes are linked - checking/unchecking a checkbox affects the text field, and at the same time, removing from the text field unchecks the checkbox.

I’m aware of the autocomplete feature, but typing it every time was tiresome for me - and in cases like services, I mostly use the `--flat` option. Alternatively, we could make it so that only the checkboxes remain, without displaying the options as text in the input

Additionally, I think an improvement could be to make the selected flags persist based on what was selected the last time - depending on the type of what is being generated. I could add this functionality, assuming it’s possible - which it most likely is.

> I would definitely up discussing a better UX approach if you want to start a discussion over that and get community feedback before implementing them. Please let me know I can certainly start one.

Sure 😉
> 
> Additionally, the `pluginUntilBuild` is set to `253.*` which is an invalid build number intellij hasn't released an IDE with that version yet this will cause error in the plugin verification stage of CI/CD. I am working on a more of an automated approach to support all future builds.

Ok, of course, I can revert the version, but are you sure it won’t work? I ran the `verifyPlugin` task, and the check passed successfully.
> 
> Now that the negatives aside. I do like the idea of adding separator in the context menu. It does provide a neat separation from rest of the menu items. Cheers for that.
> 
> Hopefully my above comments make sense. If you agree with the can update the PR accordingly? 🙌

Sure, your comments absolutely make sense. Let’s decide what to do with the improvement (checkboxes) I proposed, and I’ll make the necessary adjustments :)

I use your plugin almost every day when working with NestJS, and I really like it (great job!). As you’ve probably noticed, I’ve commented on other issues a few times before. So I decided to contribute something from myself as well, and in the process see what the development of plugins looks like since I have never dealt with it before.

dinbtechit commented:
@radoslawkoziol 
> Sure, your comments absolutely make sense. Let’s decide what to do with the improvement (checkboxes) I proposed, and I’ll make the necessary adjustments :)

I have created a discussion here to discuss the UX improvement - https://github.com/dinbtechit/jetbrains-nestjs/discussions/19. Lets document the pain points of the existing UI.

> I use your plugin almost every day when working with NestJS, and I really like it (great job!). As you’ve probably noticed, I’ve commented on other issues a few times before. So I decided to contribute something from myself as well, and in the process see what the development of plugins looks like since I have never dealt with it before.

Thanks mate. I appreciate the extra helping hand. People like you make the open source community awesome 🙌.  There are lots of things we can be working on for this plugin - Please take a look at some of the open issues and see if you can tackle any of the open issues.

radoslawkoziol commented:
@dinbtechit so what's the decision? 😅

dinbtechit commented:
I'm so sorry mate. Got affected by layoffs and trying to find a job here. Stressful times you know. Do you want to add a global settings? To turn off the checkboxes.

radoslawkoziol commented:
> I'm so sorry mate. Got affected by layoffs and trying to find a job here. Stressful times you know. Do you want to add a global settings? To turn off the checkboxes.

Oh, sorry to hear that. I hope you'll find something soon!
No worries, I'll try to work on that ;)

radoslawkoziol commented:
@dinbtechit 

So I've added NesJS settings under `Languages & Frameworks` with the checkbox we discussed (disabled by default).

Also I've added "Use custom NestJSS file icons" (enabled by default) which solves https://github.com/dinbtechit/jetbrains-nestjs/issues/6

![image](https://github.com/user-attachments/assets/8f84169e-00c7-45d6-9cd0-393b98b61dcd)

radoslawkoziol commented on gradle.properties:
```diff
@@ -4,11 +4,11 @@ pluginGroup = com.github.dinbtechit.jetbrainsnestjs
 # Supported build number ranges and IntelliJ Platform versions -> https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html
 pluginSinceBuild = 223
-pluginUntilBuild =
+pluginUntilBuild = 251.*
```
@dinbtechit I've seen you've deleted the version here in `0.0.6` but I'm afraid it won't work. And  it's not even available here 
![image](https://github.com/user-attachments/assets/666cacee-d765-4ac4-bd5b-bf977283d101)

radoslawkoziol replied:
But if  you're sure it's fine, then I'll remove it

dinbtechit replied:
Yes, I am aware of that. That's because my complete fix is not in there and also its using the older version of the gradle plugin.

radoslawkoziol replied:
So I guess we can stay with the safe option `251.*` for now

dinbtechit commented:
> @dinbtechit
> 
> So I've added NesJS settings under `Languages & Frameworks` with the checkbox we discussed (disabled by default).
> 
> Also I've added "Use custom NestJSS file icons" (enabled by default) which solves #6
> 
> ![image](https://private-user-images.githubusercontent.com/10725188/418346331-8f84169e-00c7-45d6-9cd0-393b98b61dcd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA5NTYzNzksIm5iZiI6MTc0MDk1NjA3OSwicGF0aCI6Ii8xMDcyNTE4OC80MTgzNDYzMzEtOGY4NDE2OWUtMDBjNy00NWQ2LTljZDAtMzkzYjk4YjYxZGNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzAyVDIyNTQzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5N2M5OTQwNTUwYWNjMGQ1MzJjNTJiNmIxNmJiMmY3OGE3ZThjNWIxNDU1YjExOGFkMzJhY2NjMGNiNjk2NWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Nzc6SPwy0kGAqlt3teva1e8n8EwMZB9AStIl568ZNMg)

awesome mate great job!!!!
